### PR TITLE
Feat Implement connection functionality between ICP Context and Proxy contracts

### PR DESCRIPTION
**Pull request by alenmestrov**
# Context-Proxy Contract Integration

## Summary
This PR implements the integration between context and proxy contracts, enabling secure deployment and member verification for multi-signature management.

## Key Features
- Automated proxy contract deployment through context contract
  - Context contract stores and manages proxy WASM
  - Deploys new proxy instance for each context
  - Initializes proxy with correct context ID and parent context canister ID
- Member verification system
  - Proxy contract verifies all operations with parent context
  - Checks if users are members before allowing proposal creation/approval
  - Secure cross-canister calls for verification

### Context Contract Updates
- Added proxy WASM storage and management
- Implemented `set_proxy_code` for updating proxy contract code
- Added proxy deployment logic during context creation
- Member verification endpoint for proxy contracts

### Proxy Contract Updates
- Added context contract integration
- Implemented member verification before proposal operations
- Secure initialization with context parameters

### Integration Tests
Added comprehensive test suite covering:
- Proxy deployment through context contract
- Member verification flow
- Proposal creation and approval by verified members
- Error cases for non-members

## Build and Test Instructions
Standard build and test process:
```
./build.sh
cargo test --test integration -- --nocapture
```

## Notes
- Proxy contracts are now fully integrated with their parent context
- All operations verify membership status
- Secure deployment ensures correct initialization

### CODE REVIEW

File: contracts/icp/context-config/src/mutate.rs [hunk 1]
```diff
@@ -83,12 +89,53 @@ fn add_context(
...
+    };
+
+    let (canister_record,) = create_canister(create_args, 500_000_000_000_000u128)
+        .await
+        .map_err(|e| format!("Failed to create canister: {:?}", e))?;
+
+    let canister_id = canister_record.canister_id;
+
+    // Encode init args matching the proxy's init(context_id: ICContextId, ledger_id: Principal)
+    let init_args = candid::encode_args((context_id.clone(), Principal::anonymous()))
```
Comment (miraclx) [inline contracts/icp/context-config/src/mutate.rs]: that should be the context contract, right?

```suggestion
    let init_args = candid::encode_args((context_id.clone(), configs.owner))
```

File: contracts/icp/context-config/src/mutate.rs [hunk 2]
```diff
@@ -83,12 +89,53 @@ fn add_context(
...
+            compute_allocation: None,
+            memory_allocation: None,
+            freezing_threshold: None,
+            reserved_cycles_limit: None,
+            log_visibility: None,
+            wasm_memory_limit: None,
+        }),
+    };
+
+    let (canister_record,) = create_canister(create_args, 500_000_000_000_000u128)
```
Comment (miraclx) [inline contracts/icp/context-config/src/mutate.rs:117]: should this cycle count be dynamic?

File: contracts/icp/context-config/src/mutate.rs [hunk 3]
```diff
@@ -282,3 +329,44 @@ fn revoke(
...
+    let install_args = InstallCodeArgument {
+        mode: ic_cdk::api::management_canister::main::CanisterInstallMode::Upgrade(None),
+        canister_id: proxy_canister_id,
+        wasm_module: proxy_code,
+        arg: candid::encode_one(&context_id).map_err(|e| format!("Encoding error: {}", e))?,
+    };
+
+    install_code(install_args)
+        .await
+        .map_err(|e| format!("Failed to update proxy contract: {:?}", e))?;
```
Comment (miraclx) [inline contracts/icp/context-config/src/mutate.rs:371]: but do we have permissions to do it from the context config? that seems like a massive security issue if it did

File: contracts/icp/context-config/src/mutate.rs (hunk 2 above)
Comment (alenmestrov) [reply contracts/icp/context-config/src/mutate.rs:117]: this is minimum what we need to transfer to the created contract for it to work.

File: contracts/icp/context-config/src/mutate.rs (hunk 3 above)
Comment (alenmestrov) [reply contracts/icp/context-config/src/mutate.rs:371]: yes, implemented the right way of checking the permission and preserving storage

**Review by miraclx — APPROVED** [review]

**Pull request merged**